cupy/polynomial/polynomial.py

```diff
@@ -0,0 +1,25 @@
+import cupy
+
+
+def polyvander(x, deg):
+    """Computes the Vandermonde matrix of given degree.
+
+    Args:
+        x (cupy.ndarray): array of points
+        deg (int): degree of the resulting matrix.
+
+    Returns:
+        cupy.ndarray: The Vandermonde matrix
+
+    .. seealso:: :func:`numpy.polynomial.polynomial.polyvander`
+
+    """
+    deg = cupy.polynomial.polyutils._deprecate_as_int(deg, 'deg')
+    if deg < 0:
+        raise ValueError('degree must be non-negative')
+    if x.ndim == 0:
+        x = cupy.expand_dims(x, axis=0)
+    x = cupy.expand_dims(x, axis=-1)
+    x = x + 0.0
+    v = x ** cupy.arange(deg + 1)
```

Comment:
The dtype of `cupy.arange(deg + 1)` is `int64`, so the dtype of `v` becomes always `float64` or `complex128`.
```suggestion
    dtype = numpy.float64 if x.dtype.kind in 'biu' else x.dtype
    v = x ** cupy.arange(deg + 1, dtype)
```